diff --color -Naur old/server.c new/server.c
--- old/lighttpd1.4-lighttpd-1.4.56/src/server.c	2020-11-30 05:31:14.000000000 +0800
+++ new/lighttpd1.4-lighttpd-1.4.56/src/server.c	2022-02-02 15:17:33.789539164 +0800
@@ -1547,6 +1547,7 @@
 					return -1;
 				case 0:
 					child = 1;
+					init_worker(getpid());
 					alarm(0);
 					break;
 				default:
@@ -1904,7 +1905,7 @@
 		}
 
 		connections * const joblist = connection_joblist;
-
+batch_start();
 		if (fdevent_poll(srv->ev, joblist->used ? 0 : 1000) > 0) {
 			last_active_ts = log_epoch_secs;
 		}
@@ -1914,6 +1915,7 @@
 		  : &srv->joblist_A;
 
 		server_run_con_queue(joblist);
+batch_flush();
 	}
 }
 
