--- old/nginx-1.20.0/src/os/unix/ngx_process_cycle.c	2021-04-20 21:35:47.000000000 +0800
+++ new/nginx-1.20.0/src/os/unix/ngx_process_cycle.c	2022-03-04 13:17:29.526199817 +0800
@@ -715,9 +715,9 @@
         }
 
         ngx_log_debug0(NGX_LOG_DEBUG_EVENT, cycle->log, 0, "worker cycle");
-
+batch_start();
         ngx_process_events_and_timers(cycle);
-
+batch_flush();
         if (ngx_terminate) {
             ngx_log_error(NGX_LOG_NOTICE, cycle->log, 0, "exiting");
             ngx_worker_process_exit(cycle);
