--- old/redis-7.0-rc2/src/networking.c	2022-02-28 21:35:46.000000000 +0800
+++ new/redis-7.0-rc2/src/networking.c	2022-03-14 18:36:07.198857861 +0800
@@ -1974,6 +1974,7 @@
     int processed = listLength(server.clients_pending_write);

     listRewind(server.clients_pending_write,&li);
+batch_start();
     while((ln = listNext(&li))) {
         client *c = listNodeValue(ln);
         c->flags &= ~CLIENT_PENDING_WRITE;
@@ -2008,6 +2009,7 @@
             }
         }
     }
+batch_flush();
     return processed;
 }

